### PR TITLE
charcoal pit: update loop fixes, add burn progress bar

### PR DESCRIPTION
**Pull request by qreeves**
This should fix the update loop plus make charcoal production a little saner and user friendly.

### CODE REVIEW

File: src/main/java/org/terasology/cab/system/CharcoalPitAuthoritySystem.java [hunk 1]
```diff
@@ -79,11 +79,11 @@ public void startBurningCharcoal(ProduceCharcoalRequest event, EntityRef entity,
             }
 
             int charcoalCount = CharcoalPitUtils.getResultCharcoalCount(logCount, entity);
-
-            int burnLength = 5 * 60 * 1000;
+            int burnLength = logCount * 5 * 1000;
```
Comment (MarcinSc) [inline src/main/java/org/terasology/cab/system/CharcoalPitAuthoritySystem.java]: The pit should burn a constant time, no matter how many logs. That's there point - to encourage filling it up to full.

Comment (qreeves) [inline src/main/java/org/terasology/cab/system/CharcoalPitAuthoritySystem.java]: 5 minutes is a bit excessive for a 3x3x3. Basically this just punishes the player for not creating a 99x99x99 sized pit and getting a thousand charcoal in one go. Compare this with the 16 max you get for a 3x3x3.

Comment (MarcinSc) [inline src/main/java/org/terasology/cab/system/CharcoalPitAuthoritySystem.java]: TtA is supposed to be difficult. I suggest you check out TerraFirmaCraft in Minecraft.

**Comment by Cervator** [comment]
Thanks for the fix, @qreeves ! Yeah, TTA is indeed meant to be sort of like TerraFirmaCraft, where JoshariasSurvival is more forgiving akin to normal MC or with typical mods like BC/IC :-)

One point to that is also to force the player to rely on helper creatures for bulk production - although being that we haven't implemented that yeah that does look a bit punishing for now :D

**Pull request merged**